io/src/main/java/io/smallrye/common/io/JarEntries.java

```diff
@@ -0,0 +1,13 @@
+package io.smallrye.common.io;
+
+import java.util.jar.JarEntry;
+
+public class JarEntries {
+    /**
+     * Returns the real name of this {@link JarEntry}. On Java 8, it returns the {@link JarEntry#getName()}
+     * On Java 10+, a getRealName() method was added
+     */
+    public static String getRealName(JarEntry jarEntry) {
+        return jarEntry.getName();
```

Comment:
What does this do on Java 9?  Just return a wrong answer?

Reply:
Yes, unfortunately. They realized that was missing and fixed in Java 10

Reply:
Do we want Java 9 specific bits?

When we move to 11, won't it be not required?

Reply:
I have no idea when Quarkus will move to 11, but at least this class keeps consistency until that happens

Reply:
But do we expect people to use JDK 9?

Or is it JDK 8 or 11+?

Reply:
Considering the [LTS releases](https://www.oracle.com/java/technologies/java-se-support-roadmap.html), I think it's 8 or 11+. I don't see JDK 9 as an alternative now, but that's just my opinion :wink:

Reply:
Don't forget that this is no Quarkus. I don't think it would do harm to support Java 9 if we can figure out a way to do so.

Reply:
The only way I see is using reflection (to invoke package protected methods in `JarEntry`) . Ugly but may do the job for the JDK9 users, WDYT?

Reply:
I think we don't really have to care about Java 9 and 10.

Reply:
Then we should somehow expressly prevent the library from running on 9 or 10, IMO.  I think focusing only on LTS probably isn't a good idea and sets a bad precedent, encouraging bit-rot for no good reason (i.e. avoiding 5 minutes of work).

Reply:
@Ladicek makes a good point. The project is already built in JDK 11, why do you we need to care about 9 or 10?

Reply:
Ah but `maven.compiler.target` is set to 8, ignore me :stuck_out_tongue:

Reply:
I'm not advocating LTS-only focus. At the same time, I do think it doesn't make much sense to care about non-LTS releases that are older than some existing LTS.

However, if it really is 5 minutes of work, then sure, that's better than defining an explicit support policy :-)